### ISSUE TITLE
Change how arrays/parens are handled

### ISSUE THREAD

**Issue by techniq** [techniq/odata-query#16]
After some discussion in the [PR](https://github.com/techniq/odata-query/pull/14) about how to handle `in` and parens, I think we should implement a breaking change to make array (`[ ]`) usage more consistent and easier to reason about.

## Current
Currently an array wraps each items within it in their own parens. ([code](https://github.com/techniq/odata-query/blob/master/src/index.js#L96-L100))
```js
} else if (Array.isArray(filters)) {
    const builtFilters = filters.map(f => buildFilter(f, propPrefix)).filter(f => f !== undefined);
    if (builtFilters.length) {
      return `${builtFilters.map(f => `(${f})`).join(` and `)}`
    }
```
which produces:
```js
buildFilter({ filter: [{ SomeProp: 1 }, { AnotherProp: 2 }, "startswith(Name, 'R')"] });
=> "?$filter=(SomeProp eq 1) and (AnotherProp eq 2) and (startswith(Name, 'R'))";
```

## Proposal
The proposal is to change the code to:
```js
} else if (Array.isArray(filters)) {
    const builtFilters = filters.map(f => buildFilter(f, propPrefix)).filter(f => f !== undefined);
    if (builtFilters.length) {
      return `(${builtFilters.join(` and `)})`
    }
```

which would produce:
```js
buildFilter({ filter: [{ SomeProp: 1 }, { AnotherProp: 2 }, "startswith(Name, 'R')"] });
=> "?$filter=(SomeProp eq 1 and AnotherProp eq 2 and startswith(Name, 'R'))";
```

You also do not need the individual objects for the first two (not a change, just a simplification of the example).
```js
buildFilter({ filter: [{ SomeProp: 1, AnotherProp: 2 }, "startswith(Name, 'R')"] });
=> "?$filter=(SomeProp eq 1 and AnotherProp eq 2 and startswith(Name, 'R'))";
```

If you wanted to reproduce the original results (albeit with an extra wrapping parens around it all), you would wrap each item in an array:
```js
buildFilter({ filter: [[{ SomeProp: 1 }], [{ AnotherProp: 2 }], ["startswith(Name, 'R')"]] });
=> "?$filter=((SomeProp eq 1) and (AnotherProp eq 2) and (startswith(Name, 'R')))";
```

In summary, the `[ ]` would literally translate as `( )` in the query.  If this change is made, I would release it as a new major version since it could be breaking for some users.
